### ISSUE TITLE
Python datatypes / handling of 64-bit overflow

### ISSUE THREAD

**Issue by daenuprobst** [ekzhu/datasketch#99]
As `a` and `b` have a maximum value of your chosen mersenne prime and `hv` can be any 32-bit integer, `(a * hv + b)` can yield integers larger than 2^64. Isn't that a problem? Is Python capable of dealing with this? If yes, are there performance implications?

https://github.com/ekzhu/datasketch/blob/260b80063b7d3599f93d7179a7749d626f010973/datasketch/minhash.py#L134

**Comment by ekzhu** [comment]
Even though MinHash only uses 32-bit, Python ints can be [at least 64-bit large](https://stackoverflow.com/questions/7604966/maximum-and-minimum-values-for-ints). 

The constraint here is actually numpy's integer type that has maximum 64 bit. If we were to use 64-bit hash values then, as you pointed out, `a * hv + b`will overflow. Thus the 32-bit constraint.

**Comment by daenuprobst** [comment]
Yes. But I wonder how this impacts the performance (run time) and whether it is worth letting it overflow 64 bits...

As an aside, have you considered using "fast_mod"? I use it in my C++ implementation and it speeds it up quite a lot (not sure how calling a function behaves time-wise in Python, also given that your `a` and `b` are large, I'm not sure how much gain there would be)

```
def fast_mod(x, y):
    if x < y:
        return x
    else:
        return x % y
```

**Comment by tcefrep** [comment]
Regarding the original problem, even if 32-bit MinHash is used, ` (a * hv + b)` can yield integers larger than 2^64 due to the fact that `a` and `b` are random integers ranging from 0 or 1 to (1 << 61) - 1. Will discarding the overflow affect the universality of hashing?
